src/flexspi_nor_storage_bus.rs

```
@@ -0,0 +1,1562 @@
+        self.execute_ip_cmd();
+
+        // Wait for command to complete
+        // This wait is for FlexSPI to send the command to the Flash device
+        // But the command completion in the flash needs to be checked separately by reading the status register of the flash device
+        self.wait_for_cmd_completion();
+
+        // Check for any errors during the transfer
+        if let Err(status) = self.check_transfer_status() {
+            status.describe(self);
+
+            match status {
+                FlexSpiError::AhbBusTimeout {
+                    AhbReadCmdErr: _,
+                    AhbWriteCmdErr: _,
+                } => {
+                    return Err(NorStorageBusError::StorageBusIoError);
+                }
+                FlexSpiError::CmdCheckErr {
+                    AhbReadCmdErr: _,
+                    AhbWriteCmdErr: _,
+                    IpCmdErr: _,
+                } => {
+                    return Err(NorStorageBusError::StorageBusIoError);
+                }
+                FlexSpiError::CmdExecErr {
+                    AhbReadCmdErr: _,
+                    AhbWriteCmdErr: _,
+                    IpCmdErr: _,
+                } => {
+                    return Err(NorStorageBusError::StorageBusIoError);
+                }
+                FlexSpiError::CmdGrantErr {
+                    AhbReadCmdErr: _,
+                    AhbWriteCmdErr: _,
+                    IpCmdErr: _,
+                } => {
+                    return Err(NorStorageBusError::StorageBusNotAvailable);
+                }
+                FlexSpiError::DataLearningFailed => {
+                    return Err(NorStorageBusError::StorageBusInternalError);
+                }
+            }
+        }
+
+        // For data transfer commands, read/write the data
+        if let Some(data_cmd) = cmd.cmdtype {
+            match data_cmd {
+                NorStorageCmdType::Read => {
+                    let buffer = read_buf.unwrap();
```

If there is no buffer provided, then this would panic. We should return an error.

---

fixed